### ISSUE TITLE
Cannot login with debian 12 server

### ISSUE THREAD

**Issue by anfeichtinger** [supertuxkart/stk-code#4974]
## Description

I'm currently trying to set up a new server for some friends. But I cannot seem to log in with my credentials like instructed in different tutorials. I'm stuck at the step "Log in to SuperTuxKart Online". When I run \
```supertuxkart --init-user --login="MYUSERNAME" --password="MYPASSWORD"``` 
I get a timeout. Is the api down or was it updated?

## Steps to reproduce

I have followed the steps from [here](https://gist.github.com/MarioCMFlys/617c34e7cb90e6004cd5ce0259494049). 

## Configuration

STK release version: master branch at commit "42d4eaae96b43210f6f6baa6350929168299d6e9"

STK Source (PPA, distribution package, official binary, etc.): git

System: Debian 12 running as an unprivileged container in proxmox 8.1

Graphics card: Not accessible to container / Nvidia GTX 1080

CPU: AMD Ryzen 5 1600X (virtualized with 4 cores for the container)

Gamepads/keyboards models if related to the issue: -

## Additional information

stdout.log:
```
[info   ] [FileManager]: Data files will be fetched from: '../data/'
[info   ] [FileManager]: Data files will be fetched from: '../data/../../stk-assets/'
[info   ] [FileManager]: User directory is '/home/supertuxkart/.config/supertuxkart/config-0.10/'.
[info   ] [FileManager]: Addons files will be stored in '/home/supertuxkart/.local/share/supertuxkart/addo>
[info   ] [FileManager]: Screenshots will be stored in '/home/supertuxkart/.local/share/supertuxkart/scree>
[info   ] [FileManager]: User-defined grand prix will be stored in '/home/supertuxkart/.local/share/supert>
[info   ] [FileManager]: Asset 0 will be loaded from '../data/challenges/'.
[info   ] [FileManager]: Asset 1 will be loaded from '../data/gfx/'.
[info   ] [FileManager]: Asset 2 will be loaded from '../data/grandprix/'.
[info   ] [FileManager]: Asset 3 will be loaded from '../data/gui/icons/'.
[info   ] [FileManager]: Asset 4 will be loaded from '../data/gui/screens/'.
[info   ] [FileManager]: Asset 5 will be loaded from '../data/gui/dialogs/'.
[info   ] [FileManager]: Asset 6 will be loaded from '../data/replay/'.
[info   ] [FileManager]: Asset 7 will be loaded from '../data/shaders/'.
[info   ] [FileManager]: Asset 8 will be loaded from '../data/skins/'.
[info   ] [FileManager]: Asset 9 will be loaded from '../data/ttf/'.
[info   ] [FileManager]: Asset 10 will be loaded from '../data/po/'.
[info   ] [FileManager]: Asset 11 will be loaded from '../data/../../stk-assets/library/'.
[info   ] [FileManager]: Asset 12 will be loaded from '../data/../../stk-assets/models/'.
[info   ] [FileManager]: Asset 13 will be loaded from '../data/../../stk-assets/music/'.
[info   ] [FileManager]: Asset 14 will be loaded from '../data/../../stk-assets/sfx/'.
[info   ] [FileManager]: Asset 15 will be loaded from '../data/../../stk-assets/textures/'.
[info   ] [FileManager]: Asset 16 will be loaded from '../data/../../stk-assets/tracks/'.
[info   ] GUI: Loading skin data from file: ../data/skins/peach/stkskin.xml
[info   ] GrandPrixManager: Loading Grand Prix files from ../data/grandprix/
[info   ] GrandPrixManager: Loading Grand Prix files from /home/supertuxkart/.local/share/supertuxkart/gra>
[info   ] HTTPRequest: Sending username=MYUSERNAME&password=*************&save-session=true to https://o>
[error  ] Main: Timed out trying login, check login info or connection to stk addons.
[info   ] Singleton: Destroyed singleton.
```

Error message:

![image](https://github.com/supertuxkart/stk-code/assets/80401865/f2f05278-3f8b-47b5-b247-326183bcbbbf)

**Comment by Lulu-Games** [comment]
Does your Server work properly? Have you enabled port forwarding on your Router?
Do you already have an account for STK-online?

**Comment by anfeichtinger** [comment]
> Does your Server work properly? Have you enabled port forwarding on your Router? Do you already have an account for STK-online?

Yes my server does work properly, I'm running multiple things like websites, services and other game servers without any problem. I can also ping ```online.supertuxkart.net``` from the container, so it has access to the internet. (Of course I also was able to clone the repo.) 

![grafik](https://github.com/supertuxkart/stk-code/assets/80401865/14903cc5-89c5-4474-994a-665f9cf6359f)

I did not set up port forwarding on my router yet, usually I test everything locally before doing that. The proxmox server is in my local network. As the error occurs during a normal http request this should not be an issue in my case but correct me if I'm wrong.

And finally, yes I did create an account for STK-online and I can use it just fine in the game an on the website.

**Comment by Lulu-Games** [comment]
Your ping latency is very high. That may cause Problems (over 100ms). Your error code says "connection timed out" that could mean, the latency is the Problem. If you only test it in local network, Port forwarding shouldnt be a Problem. And isn't online.supertuxkart.net https not http?

**Comment by Lulu-Games** [comment]
And how long does it take, if the error message from the server is printed, after the  message "HttpRequest: Sending username=****&password=**** to https://online.supertuxkart.net". Depending on that it's a latency problem or any other problem.

**Comment by Benau** [comment]
You can just copy players.xml in stk config folder from a working stk installation to your server, make sure your working stk has save password on

**Comment by Lulu-Games** [comment]
But if there is an issue with the Server itself that won't work

**Comment by anfeichtinger** [comment]
> Your ping latency is very high. That may cause Problems (over 100ms). Your error code says "connection timed out" that could mean, the latency is the Problem. If you only test it in local network, Port forwarding shouldnt be a Problem. And isn't online.supertuxkart.net https not http?

Yes, unfortunately I don't have a wired connection to my home. It was never a problem for other game server though, there the ping also is almost always under 100ms. The request is sent via the following command: ```supertuxkart --init-user --login="MYUSERNAME" --password="MYPASSWORD"``` which decides how the request is sent so https or http should not be a problem on my side.

> And how long does it take, if the error message from the server is printed, after the message "HttpRequest: Sending username=**&password=** to https://online.supertuxkart.net". Depending on that it's a latency problem or any other problem. And have you built the server from the source code, or have you used the prebuilt binary (if there is any)?

It takes around 20 seconds so the 100ms ping should not be the cause for that. I have built the server from source (without specifying a version number) following the steps in [this tutorial](https://gist.github.com/MarioCMFlys/617c34e7cb90e6004cd5ce0259494049). If there is a better approach please let me know.

> You can just copy players.xml in stk config folder from a working stk installation to your server, make sure your working stk has save password on

Thank you, this did the trick. I am now able to connect to the server from my Windows install!
But that still doesn't explain the timeout issue. Should I keep this open in order to find a solution or close it for now?

**Comment by Lulu-Games** [comment]
You can keep it open. Isn't the timeout issue the Main issue?

**Comment by Lulu-Games** [comment]
You are running the Server on a virtzal Maschine, don't you? Maybe the VM blocks the connection

**Comment by anfeichtinger** [comment]
Yes, the timeout is the main issue. 

Yes, the server is running in an lxc container on proxmox. But I don't see why the connection should be blocked since every other server with the same configuration runs fine and other requests in this container also work fine. I also tried disabling the firewall for the container with no luck.

**Comment by Lulu-Games** [comment]
In your shoes I would clone the repo again and recompile

**Comment by anfeichtinger** [comment]
I already did that three times, always with the same timeout. But with the solution of @Benau I was able to get everything running.

I copied the players.xml file from my logged in Windows instance at ```C:\Users\USERNAME\AppData\Roaming\supertuxkart\config-0.10``` to the server at ```/home/USERNAME/.config/supertuxkart/config-0.10/players.xml``` (via ftp) and skipped the step. 

Everything running smoothly now except for this warning in the logs which doesn't seem to cause problems so far. 
```[warn   ] STKHost: Failed to get enough stun servers using SRV record.```

**Comment by qwertychouskie** [comment]
`[info   ] HTTPRequest: Sending username=MYUSERNAME&password=*************&save-session=true to https://o>`

You have to actually give a valid username/password

**Comment by Lulu-Games** [comment]
Yes, but that was actually an example because of privacy reasons

**Comment by searinminecraft** [comment]
could you try to send a request with cURL and see if it succeeds?

`curl -X POST -d "username=MyUsername&password=Password&save-session=true" https://online.supertuxkart.net/api/v2/user/connect`
